.github/workflows/test.yml

```diff
@@ -43,10 +55,8 @@ jobs:
     - uses: actions/setup-python@v4
       with:
         python-version: '3.9' # As defined in tests/scripts/docker/run_tests.sh
-        cache: 'pip'          # caching pip dependencies
-
-    - run: pip install -r requirements-tests.txt
-    - run: pip install -r requirements.txt
+    - run: python -m pip install --upgrade pip
```

Comment:
This might help with the weirdness flakiness when running pip install pretty much after the setting up python3.9